examples/common/pigweed/rpc_services/Attributes.h

```diff
@@ -33,14 +33,35 @@
 #include <app/data-model-provider/Provider.h>
 #include <app/util/attribute-storage.h>
 #include <app/util/attribute-table.h>
+#include <data-model-providers/codegen/CodegenDataModelProvider.h>
 #include <lib/core/TLV.h>
 #include <lib/core/TLVTags.h>
 #include <lib/core/TLVTypes.h>
+#include <pigweed/rpc_services/AccessInterceptor.h>
+#include <pigweed/rpc_services/AccessInterceptorRegistry.h>
 #include <platform/PlatformManager.h>
+#include <set>
 
 namespace chip {
 namespace rpc {
 
+std::optional<::pw::Status> TryWriteViaAccessor(const chip::app::ConcreteDataAttributePath & path,
+                                                chip::app::AttributeValueDecoder & decoder)
+{
+    std::set<PigweedDebugAccessInterceptor *> accessors = PigweedDebugAccessInterceptorRegistry::Instance().GetAllAccessors();
+
+    for (PigweedDebugAccessInterceptor * accessor : accessors)
+    {
+        std::optional<::pw::Status> result = accessor->Write(path, decoder);
+        if (result.has_value()) // Write was either a success or failure.
+        {
+            return result;
+        }
+    }
+
+    return std::nullopt;
```

Comment:
would it make sense to assert that decoder was not decoded from, to avoid hard to track bugs because our existing implementation depends on that state?